src/IotTelemetrySimulator/TelemetryValues.cs

```diff
@@ -123,26 +130,21 @@ public TelemetryValues(IList<TelemetryVariable> variables)
         /// Removes non-used variables in a sequence.
         /// This way we can keep the a counter variable incrementally correctly if the sequence did not use it in current iteration.
         /// </summary>
-        private static void ResetNonUsedReferencedVariables(
+        private static void ResetNotUsedReferencedVariables(
             Dictionary<string, object> previous,
             Dictionary<string, object> next,
-            TelemetryVariable sequenceVariable,
-            string usedVariable)
+            IEnumerable<string> notUserVariables)
         {
-            var referencedVariables = sequenceVariable.GetReferenceVariableNames();
-            foreach (var referencedVariable in referencedVariables)
+            foreach (var notUsedVariable in notUserVariables)
             {
-                if (referencedVariable != usedVariable)
+                // Restore it from the previous value.
+                if (previous != null && previous.TryGetValue(notUsedVariable, out var previousValue))
                 {
-                    // Restore it from the previous value.
-                    if (previous != null && previous.TryGetValue(referencedVariable, out var previousValue))
-                    {
-                        next[referencedVariable] = previousValue;
-                    }
-                    else
-                    {
-                        next.Remove(referencedVariable);
-                    }
+                    next[notUsedVariable] = previousValue;
+                }
+                else
+                {
+                    next.Remove(notUsedVariable);
```

Comment:
Can `next` contain `notUsedVariable` at all?

Reply:
It does:
vars:
- counter1
- counter2
- sequence [$.counter1, $counter2]

In first iteration:
[ { $.counter1, 1 }, { $.counter2, 1 } ] => we remove $.counter2 from next so that it can start from 1 (or min).

2nd iteration: 
[ { $.counter1, 1 }, { $.counter2, 1 } ] => we get $.counter1 from prev